src/Workspaces/Core/Portable/Options/SolutionOptionSet.cs

```diff
@@ -56,15 +49,24 @@ internal SolutionOptionSet(ILegacyWorkspaceOptionService globalOptions)
         {
             if (_values.TryGetValue(optionKey, out var value))
             {
-                return value is ICodeStyleOption codeStyleOption ? codeStyleOption.AsPublicCodeStyleOption() : value;
+                return value;
             }
 
-            value = _globalOptions.GetOption(optionKey);
+            // Global options store internal representation of code style options. Translate to public representation.
+            var internalValue = _globalOptions.GetOption(optionKey);
```

Comment:
nit - rename `_globalOptions` to better match type

Reply:
Yes, `SolutionOptionSet` is deprecated. `LegacyWorkspaceOptionService` is there only to support the existing public API semantics, which is that `Solution.Options` (which is backed by `SolutionOptionSet`) read values from global options and `Workspace.TryApplyChanges` pushes option updates to all registered workspaces.